sources/packages/backend/libs/services/src/notifications/notification/notification.service.ts

```diff
@@ -55,17 +56,26 @@ export class NotificationService extends RecordDataModelService<Notification> {
     notifications: SaveNotificationModel[],
     auditUserId: number,
     options?: {
-      entityManager: EntityManager;
+      disbursementId?: number;
+      entityManager?: EntityManager;
     },
   ): Promise<number[]> {
-    const newNotifications = notifications.map((notification) => ({
-      user: { id: notification.userId } as User,
-      creator: { id: auditUserId } as User,
-      messagePayload: notification.messagePayload,
-      notificationMessage: {
-        id: notification.messageType,
-      } as NotificationMessage,
-    }));
+    const newNotifications = notifications.map((notification) => {
+      const notificationToSave = {
+        user: { id: notification.userId } as User,
+        creator: { id: auditUserId } as User,
+        messagePayload: notification.messagePayload,
+        notificationMessage: {
+          id: notification.messageType,
+        } as NotificationMessage,
+      };
+      if (options?.disbursementId) {
+        notificationToSave["metadata"] = {
+          disbursement_id: options?.disbursementId,
+        };
```

Comment:
Minor. Maybe we can do `notificationToSave.metadata = { ... }` instead of `notificationToSave["metadata"] = { ... }`. If we do the former style, we should probably add `metadata` as optional field to the interface [`SaveNotificationModel`](https://github.com/bcgov/SIMS/blob/52dd32d09387a7b82dfc2d86df7e24c46b8550c6/sources/packages/backend/libs/services/src/notifications/notification/notification.model.ts#L4). Thoughts @dheepak-aot @andrewsignori-aot ?